hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -1656,6 +1686,21 @@ func (r *NodePoolReconciler) enqueueNodePoolsForConfig(obj client.Object) []reco
 	return result
 }
 
+// getNodePoolNamespace returns the namespaced name of a NodePool, given the NodePools name
+// and the control plane namespace name for the hosted cluster that this NodePool is a part of.
+func (r *NodePoolReconciler) getNodePoolNamespacedName(nodePoolName string, controlPlaneNamespace string) (types.NamespacedName, error) {
```

Comment:
can we please unit test this?

Reply:
Thanks @enxebre, I added a unit test for this function.